db/event_manager_test.go

```diff
@@ -275,72 +273,125 @@ func TestUnhandledEvent(t *testing.T) {
 
 }
 
-func InitWebhookTest() (*int, *float64, *[][]byte) {
-
-	// Uses counter and sum values for simplified tracking of POST requests recieved by HTTP
-	// TODO:  enhance by adding listener for /count, /sum, /reset endpoints, and leave
-	// all management within the function, instead of sharing pointer references around
-	counter := 0
-	sum := 0.0
-	var payloads [][]byte
-	// Start HTTP listener for webhook calls
-	go func() {
-		http.HandleFunc("/slow", func(w http.ResponseWriter, r *http.Request) {
-			time.Sleep(1000 * time.Millisecond)
-			counter++
-			fmt.Fprintf(w, "OK")
-		})
-		http.HandleFunc("/slow_2s", func(w http.ResponseWriter, r *http.Request) {
-			time.Sleep(2 * time.Second)
-			counter++
-			fmt.Fprintf(w, "OK")
-		})
-		http.HandleFunc("/slow_5s", func(w http.ResponseWriter, r *http.Request) {
-			time.Sleep(5 * time.Second)
-			counter++
-			fmt.Fprintf(w, "OK")
-		})
-		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-			r.ParseForm()
-
-			body, err := ioutil.ReadAll(r.Body)
+// Uses WebhookRequest for simplified tracking of POST requests received by HTTP.
+// A mutex has been embedded in WebhookRequest to avoid race conditions.
+type WebhookRequest struct {
+	mutex    sync.Mutex
+	count    int
+	sum      float64
+	payloads [][]byte
+}
+
+func (wr *WebhookRequest) GetCount() int {
+	wr.mutex.Lock()
+	defer wr.mutex.Unlock()
```

Comment:
Thank you Adam.  I thought using defer would reduce one line of code and the code looks more clean. But you're absolutely right it's not free. I've removed the defer statement to avoid scheduler overhead.

```
func (wr *WebhookRequest) GetCount() int {
	wr.mutex.Lock()
	count := wr.count
	wr.mutex.Unlock()
	return count
}
```